apps/exchange/test/matching_engine_test.exs

```diff
@@ -656,4 +656,152 @@ defmodule MatchingEngineTest do
+        end)
+        |> Enum.map(fn {_cast_event, _event, payload} ->
+          payload.order_id
+        end)
+      trade_ids =
+        TestEventBus.value()
+        |> Enum.filter(fn {_cast_event, event, _payload} ->
+          event == :trade_executed
+        end)
+        |> Enum.map(fn {_cast_event, _event, payload} ->
+          {payload.buyer_id , payload.seller_id, payload.buy_order_id, payload.sell_order_id}
+        end)
+
+      {buyer_id , seller_id, buy_order_id, sell_order_id} = hd(trade_ids)
+      assert order_queued_ids == ids
+      assert Enum.count(order_queued_ids) == 2
+      assert buyer_id == "alchemist0"
+      assert seller_id == "alchemist9"
+      assert sell_order_id == "9"
+      assert buy_order_id == "100"
+      assert Enum.count(trade_ids) == 1
+    end
+
+    test "Check if order_cancelled event is correctly broadcasted", %{order_book: order_book} do
+      TestEventBus.flush()
+      order_1 = Utils.sample_order(%{size: 1000, price: 1000, side: :buy})
+      order_2 = Utils.sample_order(%{size: 1000, price: 2000, side: :sell})
+      order_1 = %Order{order_1 | trader_id: "alchemist0", order_id: "100"}
+      order_2 = %Order{order_2 | trader_id: "alchemist9"}
+      ids = ~w(100 9)
+      {_reply, _response, order_book} =
+        MatchingEngine.handle_call({:place_limit_order, order_1}, nil, order_book)
+      {_reply, _response, order_book} =
+        MatchingEngine.handle_call({:place_limit_order, order_2}, nil, order_book)
+
+      {_reply, _response, order_book} =
+        MatchingEngine.handle_call({:cancel_order, "9"}, nil, order_book)
+      {_reply, _response, _order_book} =
+        MatchingEngine.handle_call({:cancel_order, "100"}, nil, order_book)
+
+      order_queued_ids =
+        TestEventBus.value()
+        |> Enum.filter(fn {_cast_event, event, _payload} ->
+          event == :order_queued
+        end)
+        |> Enum.map(fn {_cast_event, _event, payload} ->
+          payload.order_id
+        end)
+      cancel_ids =
+        TestEventBus.value()
+        |> Enum.filter(fn {_cast_event, event, _payload} ->
+          event == :order_cancelled
+        end)
+        |> Enum.map(fn {_cast_event, _event, payload} ->
+          payload.order_id
+        end)
+      assert order_queued_ids == ids
+      assert Enum.count(order_queued_ids) == 2
+      assert cancel_ids == Enum.reverse(ids)
+      assert Enum.count(cancel_ids) == 2
+    end
+
+    test "Check if order_expired event is correctly broadcasted", %{order_book: order_book} do
+      TestEventBus.flush()
+      order_1 = Utils.sample_order(%{size: 1000, price: 1000, side: :buy})
+      t = :os.system_time(:millisecond) - 2000
+      order_1 = %Order{order_1 | trader_id: "alchemist0", order_id: "100", exp_time: t}
+
+      {_reply, _response, order_book} =
+        MatchingEngine.handle_call({:place_limit_order, order_1}, nil, order_book)
+      {:noreply, _order_book} = MatchingEngine.handle_info(:check_expiration, order_book)
```

Comment:
Shouldn't  call it directly, same as `handle_call`. Instead we can find the `pid `and send a `:check_expiration `message